tests/integration/suite/test_workloads.py

```diff
@@ -341,6 +341,49 @@ def _timestamp_reset():
              fail_handler=lambda: 'Timed out waiting for timestamp reset')
 
 
+def test_perform_workload_action_read_only(admin_mc, admin_pc,
+                                           remove_resource, user_mc):
+    client = admin_pc.client
+    project = admin_pc.project
+    user = user_mc
+    remove_resource(user)
+
+    ns = admin_pc.cluster.client.create_namespace(
+        name=random_str(),
+        projectId=project.id)
+    remove_resource(ns)
+
+    prtb = admin_mc.client.create_project_role_template_binding(
+        name="prtb-" + random_str(),
+        userId=user.user.id,
+        projectId=project.id,
+        roleTemplateId="read-only")
+    remove_resource(prtb)
+
+    wait_until_available(user.client, project)
+
+    workload_name = random_str()
+    workload = client.create_workload(
+        name=workload_name,
+        namespaceId=ns.id,
+        scale=1,
+        containers=[{
+            'name': 'one',
+            'image': 'nginx',
+        }])
+    remove_resource(workload)
+
+    wait_for_workload(client, workload_name)
+
+    with pytest.raises(ApiError) as e:
+        user.client.action(
+            obj=workload,
+            action_name="rollback",
+            answers={"asdf": "asdf"})
+    assert e.value.error.status == 404
+    client.delete(ns)
```

Comment:
are we removing this since its already removed above on line 354 ?

Reply:
Yes, good catch.